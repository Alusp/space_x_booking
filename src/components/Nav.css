body {
  padding: 2%;
}

.navig-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  align-content: center;
}

.image {
  width: 10%;
  height: 45%;
}

.logo-header {
  display: flex;
}

.title {
  padding-left: 5%;
}

.navBar {
  display: flex;
  grid-gap: 25%;
  list-style: none;
  padding-right: 10%;
  text-decoration: none;
}

/* for noneactiv router */

ul {
  list-style: none;
  display: flex;
  justify-content: space-around;
}

ul li a {
  text-decoration: none;
  padding: 10px 20px;
  font-weight: bold;
}

.active {
  text-decoration: underline;
}

.active a {
  text-decoration: underline;
}
