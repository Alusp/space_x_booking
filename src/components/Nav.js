import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import Planet from '../images/planet.png';
import './Nav.css';

const Nav = () => {
  const location = useLocation();

  const { pathname } = location;

  const splitLocation = pathname.split('/');
  return (

    <>
      <div className="navig-container">
        <div className="logo-header">
          <img alt="logo" className="image" src={Planet} />
          <h1 className="title">Space Travelers Hub</h1>
        </div>
        <ul>
          <li className={splitLocation[1] === '' ? 'active' : ''}>
            <Link to="/">Rockets</Link>
          </li>
          <li className={splitLocation[1] === 'mission' ? 'active' : ''}>
            <Link to="/mission">Mission</Link>
          </li>
          <li className={splitLocation[1] === 'myprofile' ? 'active' : ''}>
            <Link to="/myprofile">Profile</Link>
          </li>
        </ul>
      </div>
    </>
  );
};
export default Nav;
