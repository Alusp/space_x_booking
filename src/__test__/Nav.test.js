import { MemoryRouter as Router } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';
import '@testing-library/jest-dom';

it('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <Nav />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
