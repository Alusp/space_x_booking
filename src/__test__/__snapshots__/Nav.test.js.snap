// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navig-container"
>
  <div
    className="logo-header"
  >
    <img
      alt="logo"
      className="image"
      src="planet.png"
    />
    <h1
      className="title"
    >
      Space Travelers Hub
    </h1>
  </div>
  <ul>
    <li
      className="active"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/mission"
        onClick={[Function]}
      >
        Mission
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/myprofile"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
</div>
`;
